.github/workflows/pre-post-release.yml

```
@@ -0,0 +1,97 @@
+---
+name: Pre/Post Release
+
+on:
+  workflow_call:
+    ref:
+      description: 'Branch or tag ref to run the workflow on'
+      required: true
+      default: 'main'
+    version:
+      description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+      required: true
+    phase:
+      description: 'Pre or post release phase'
+      type: choice
+      options:
+        - pre
+        - post
+      required: true
+
+env:
+  RELEASE_VERSION: ${{ inputs.version }}
+  BRANCH_NAME: ${{ inputs.phase }}-release-v${{ inputs.version }}
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  validate-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Validate tag does not exist on current commit
+        uses: ./.github/workflows/validate-tag
+        with:
+          tag: ${{ env.RELEASE_VERSION }}
+
+  create-pr:
+    name: "Bump versions and create PR"
+    runs-on: ubuntu-latest
+    needs:
+      - validate-tag
+    permissions:
+      contents: write
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+        with:
+          username: ${{ env.GIT_USER }}
+          email: ${{ env.GIT_EMAIL }}
+          token: ${{ env.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          token: ${{ env.GITHUB_TOKEN }}
+
+      - name: Create the release tag (post phase)
+        if: inputs.phase == 'post'
+        run: |
+          git tag "v${{ env.RELEASE_VERSION }}"
+          git push origin "v${{ env.RELEASE_VERSION }}"
+
+      - name: Create a ${{ inputs.phase }} release branch
+        run: git checkout -b ${{ env.BRANCH_NAME }}
+
+      - name: Pre bump versions (pre phase)
```

[minor] we can reuse the wording closer to maven release plugin that we are used to
```suggestion
      - name: Set release version (pre release)
```

---

Ref: https://github.com/elastic/ecs-logging-java/pull/226/commits/8648221f313b4bdba9fa09d96046fba28783dbba